### PR TITLE
Fix API integration tests

### PR DESCRIPTION
**Pull request by Selutario**
## Description
Hi team!

This PR fixes the API integration tests that were failing in 4.2.

## Tests
```
root@wazuh-master:/# /var/ossec/framework/python/bin/python3 -m pytest /wazuh-qa/tests/integration/test_api
/var/ossec/framework/python/lib/python3.9/site-packages/_testinfra_renamed.py:5: DeprecationWarning: testinfra package has been renamed to pytest-testinfra. Please `pip install pytest-testinfra` and `pip uninstall testinfra` and update your package requirements to avoid this message
  warnings.warn((
====================================================================================== test session starts ======================================================================================
platform linux -- Python 3.9.2, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /wazuh-qa/tests/integration, configfile: pytest.ini
plugins: html-3.1.1, testinfra-6.3.0, testinfra-6.0.0, metadata-1.11.0
collected 78 items                                                                                                                                                                              

wazuh-qa/tests/integration/test_api/test_config/test_DOS_blocking_system/test_DOS_blocking_system.py .ss.                                                                                 [  5%]
wazuh-qa/tests/integration/test_api/test_config/test_bruteforce_blocking_system/test_bruteforce_blocking_system.py .ss.                                                                   [ 10%]
wazuh-qa/tests/integration/test_api/test_config/test_cache/test_cache.py .ss.                                                                                                             [ 15%]
wazuh-qa/tests/integration/test_api/test_config/test_cors/test_cors.py xX                                                                                                                 [ 17%]
wazuh-qa/tests/integration/test_api/test_config/test_drop_privileges/test_drop_privileges.py .ss.                                                                                         [ 23%]
wazuh-qa/tests/integration/test_api/test_config/test_experimental_features/test_experimental_features.py .ss.                                                                             [ 28%]
wazuh-qa/tests/integration/test_api/test_config/test_host_port/test_host_port.py .s.ss.s.                                                                                                 [ 38%]
wazuh-qa/tests/integration/test_api/test_config/test_https/test_https.py .ss.                                                                                                             [ 43%]
wazuh-qa/tests/integration/test_api/test_config/test_jwt_token_exp_timeout/test_jwt_token_exp_timeout.py .ss.                                                                             [ 48%]
wazuh-qa/tests/integration/test_api/test_config/test_logs/test_logs.py .ss.                                                                                                               [ 53%]
wazuh-qa/tests/integration/test_api/test_config/test_rbac/test_rbac_mode.py .ss.                                                                                                          [ 58%]
wazuh-qa/tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py .s.s.s.ss.s.s.s.                                                                               [ 79%]
wazuh-qa/tests/integration/test_api/test_rbac/test_add_old_resource.py ....                                                                                                               [ 84%]
wazuh-qa/tests/integration/test_api/test_rbac/test_admin_resources.py ..FF                                                                                                                [ 89%]
wazuh-qa/tests/integration/test_api/test_rbac/test_policy_position.py E                                                                                                                   [ 91%]
wazuh-qa/tests/integration/test_api/test_rbac/test_remove_relationship.py EEE                                                                                                             [ 94%]
wazuh-qa/tests/integration/test_api/test_rbac/test_remove_resource.py EEEE                                                                                                                [100%]
```

RBAC tests are failing due to:
```JSON
Error obtaining login token: 
{
    "title": "Too Many Requests",
    "detail": "Maximum number of request per minute reached",
    "remediation": "This limit can be changed in security.yaml file. More information here: https://documentation.wazuh.com/current/user-manual/api/configuration.html#configuration-file",
    "error": 6001,
}
```

It is strange because the API is restarted before each test. It doesn't seem to happen when separately running the `test_rbac`. I'm still investigaing it. 

Regards,
Selu.

### CODE REVIEW

**Comment by Selutario** [comment]
## Bug found (and fixed)

Hey team!

Finally, I found what was making the RBAC tests fail when run after the API config ones. If we look at the test execution order:
```
wazuh-qa/tests/integration/test_api/test_config/test_DOS_blocking_system/test_DOS_blocking_system.py .ss.
wazuh-qa/tests/integration/test_api/test_config/test_bruteforce_blocking_system/test_bruteforce_blocking_system.py .ss.
...
wazuh-qa/tests/integration/test_api/test_rbac/test_add_old_resource.py ....
wazuh-qa/tests/integration/test_api/test_rbac/test_admin_resources.py ..FF
```

we see that the first one is `test_DOS_blocking_system`. Said test changes the API config to add a reduced number of requests_per_minute to the API:
```YAML
---
- tags:
  - config1
  configuration:
    access:
      max_request_per_minute: 50
- tags:
  - config2
  configuration:
    access:
      max_request_per_minute: 10
```

Although there is a mechanism to restore the original API configuration after each test:
https://github.com/wazuh/wazuh-qa/blob/28d14fe3e62d9bd5d846b0bc046d11278a01cfcc/tests/integration/test_api/conftest.py#L57-L59

It turns out that, since `test_DOS_blocking_system` is the first test that is executed, the `api.yaml` file was empty before starting. Therefore, the configuration was changed and not later restored.

Then, rbac tests make many requests to the API which still has `max_request_per_minute: 50` so the error described in the PR is raised. 

## Tests output
This has been fixed and it is running fine now:
```
pytest /wazuh-qa/tests/integration/test_api
/var/ossec/framework/python/lib/python3.9/site-packages/_testinfra_renamed.py:5: DeprecationWarning: testinfra package has been renamed to pytest-testinfra. Please `pip install pytest-testinfra` and `pip uninstall testinfra` and update your package requirements to avoid this message
  warnings.warn((
====================================================================================== test session starts ======================================================================================
platform linux -- Python 3.9.2, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /wazuh-qa/tests/integration, configfile: pytest.ini
plugins: html-3.1.1, testinfra-6.3.0, testinfra-6.0.0, metadata-1.11.0
collected 78 items                                                                                                                                                                              

wazuh-qa/tests/integration/test_api/test_config/test_DOS_blocking_system/test_DOS_blocking_system.py .ss.                                                                                 [  5%]
wazuh-qa/tests/integration/test_api/test_config/test_bruteforce_blocking_system/test_bruteforce_blocking_system.py .ss.                                                                   [ 10%]
wazuh-qa/tests/integration/test_api/test_config/test_cache/test_cache.py .ss.                                                                                                             [ 15%]
wazuh-qa/tests/integration/test_api/test_config/test_cors/test_cors.py xX                                                                                                                 [ 17%]
wazuh-qa/tests/integration/test_api/test_config/test_drop_privileges/test_drop_privileges.py .ss.                                                                                         [ 23%]
wazuh-qa/tests/integration/test_api/test_config/test_experimental_features/test_experimental_features.py .ss.                                                                             [ 28%]
wazuh-qa/tests/integration/test_api/test_config/test_host_port/test_host_port.py .s.ss.s.                                                                                                 [ 38%]
wazuh-qa/tests/integration/test_api/test_config/test_https/test_https.py .ss.                                                                                                             [ 43%]
wazuh-qa/tests/integration/test_api/test_config/test_jwt_token_exp_timeout/test_jwt_token_exp_timeout.py .ss.                                                                             [ 48%]
wazuh-qa/tests/integration/test_api/test_config/test_logs/test_logs.py .ss.                                                                                                               [ 53%]
wazuh-qa/tests/integration/test_api/test_config/test_rbac/test_rbac_mode.py .ss.                                                                                                          [ 58%]
wazuh-qa/tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py .s.s.s.ss.s.s.s.                                                                               [ 79%]
wazuh-qa/tests/integration/test_api/test_rbac/test_add_old_resource.py ....                                                                                                               [ 84%]
wazuh-qa/tests/integration/test_api/test_rbac/test_admin_resources.py ....                                                                                                                [ 89%]
wazuh-qa/tests/integration/test_api/test_rbac/test_policy_position.py .                                                                                                                   [ 91%]
wazuh-qa/tests/integration/test_api/test_rbac/test_remove_relationship.py ...                                                                                                             [ 94%]
wazuh-qa/tests/integration/test_api/test_rbac/test_remove_resource.py ....                                                                                                                [100%]

-- Docs: https://docs.pytest.org/en/stable/warnings.html
========================================================= 46 passed, 30 skipped, 1 xfailed, 1 xpassed, 43 warnings in 808.23s (0:13:28) =========================================================
```

Regards,
Selu.

File: tests/integration/test_api/test_config/test_bruteforce_blocking_system/test_bruteforce_blocking_system.py [hunk 1]
```diff
@@ -50,9 +50,8 @@ def test_bruteforce_blocking_system(tags_to_apply, get_configuration, configure_
     max_login_attempts = get_configuration['configuration']['access']['max_login_attempts']
 
     # Provoke a block from an unknown IP ('max_login_attempts' attempts with incorrect credentials).
-    for _ in range(max_login_attempts):
-        with pytest.raises(Exception):
-            get_api_details(user='wrong', password='wrong')
+    with pytest.raises(Exception):
```
Comment (Rebits) [inline tests/integration/test_api/test_config/test_bruteforce_blocking_system/test_bruteforce_blocking_system.py:53]: Maybe is better to use a less generic exception.

```suggestion
    with pytest.raises(RunTimeError):
```

**Review by Rebits — COMMENTED** [review]
GJ, this looks to fix all the error related with api integration tests: https://ci.wazuh.info/job/Test_integration/2499/console.

Due to it is necessary to document all tests, as they have been modified, it would be desirable to update the docstrings to follow repository standards.

File: tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py [hunk 2]
```diff
@@ -77,11 +77,11 @@ def test_add_agent(tags_to_apply, get_configuration, configure_api_environment,
 
     # Assert if an error code was returned when wazuh-authd is disabled and use_only_authd enabled.
     if use_only_authd:
-        assert post_response.status_code == 500, 'Expected status code was 500, ' \
-                                                 f'but {post_response.status_code} was returned. \nFull response: {post_response.text}'
+        assert post_response.status_code == 400, 'Expected status code was 400, but {post_response.status_code} was ' \
+                                                 'returned. \nFull response: {post_response.text}'
```
Comment (Rebits) [inline tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py:81]:
```suggestion
        assert post_response.status_code == 400, f'Expected status code was 400, but {post_response.status_code} was ' \
                                                 f'returned. \nFull response: {post_response.text}'
```

File: tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py [hunk 3]
```diff
@@ -77,11 +77,11 @@ def test_add_agent(tags_to_apply, get_configuration, configure_api_environment,
...
     else:
-        assert post_response.status_code == 200, 'Expected status code was 200, ' \
-                                                 f'but {post_response.status_code} was returned. \nFull response: {post_response.text}'
+        assert post_response.status_code == 200, 'Expected status code was 200, but {post_response.status_code} was ' \
+                                                 'returned. \nFull response: {post_response.text}'
```
Comment (Rebits) [inline tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py:84]:
```suggestion
        assert post_response.status_code == 200, f'Expected status code was 200, but {post_response.status_code} was ' \
                                                f'returned. \nFull response: {post_response.text}'
```

File: tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py [hunk 4]
```diff
@@ -123,11 +123,11 @@ def test_insert_agent(tags_to_apply, get_configuration, configure_api_environmen
 
     # Assert if an error code was returned when wazuh-authd is disabled and use_only_authd enabled.
     if use_only_authd:
-        assert post_response.status_code == 500, 'Expected status code was 500, ' \
-                                                 f'but {post_response.status_code} was returned. \nFull response: {post_response.text}'
+        assert post_response.status_code == 400, 'Expected status code was 400, but {post_response.status_code} was ' \
+                                                 'returned. \nFull response: {post_response.text}'
```
Comment (Rebits) [inline tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py:127]:
```suggestion
        assert post_response.status_code == 400, f'Expected status code was 400, but {post_response.status_code} was ' \
                                                 f'returned. \nFull response: {post_response.text}'
```

File: tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py [hunk 5]
```diff
@@ -123,11 +123,11 @@ def test_insert_agent(tags_to_apply, get_configuration, configure_api_environmen
...
     else:
-        assert post_response.status_code == 200, 'Expected status code was 200, ' \
-                                                 f'but {post_response.status_code} was returned. \nFull response: {post_response.text}'
+        assert post_response.status_code == 200, 'Expected status code was 200, but {post_response.status_code} was ' \
+                                                 'returned. \nFull response: {post_response.text}'
```
Comment (Rebits) [inline tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py:130]:
```suggestion
        assert post_response.status_code == 200, f'Expected status code was 200, but {post_response.status_code} was ' \
                                                f'returned. \nFull response: {post_response.text}'
```

File: tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py [hunk 6]
```diff
@@ -164,11 +164,11 @@ def test_insert_quick_agent(tags_to_apply, get_configuration, configure_api_envi
 
     # Assert if an error code was returned when wazuh-authd is disabled and use_only_authd enabled.
     if use_only_authd:
-        assert post_response.status_code == 500, 'Expected status code was 500, ' \
-                                                 f'but {post_response.status_code} was returned. \nFull response: {post_response.text}'
+        assert post_response.status_code == 400, 'Expected status code was 400, but {post_response.status_code} was ' \
+                                                 'returned. \nFull response: {post_response.text}'
```
Comment (Rebits) [inline tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py:168]:
```suggestion
        assert post_response.status_code == 400, f'Expected status code was 400, but {post_response.status_code} was ' \
                                                 f'returned. \nFull response: {post_response.text}'
```

File: tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py [hunk 7]
```diff
@@ -164,11 +164,11 @@ def test_insert_quick_agent(tags_to_apply, get_configuration, configure_api_envi
...
     else:
-        assert post_response.status_code == 200, 'Expected status code was 200, ' \
-                                                 f'but {post_response.status_code} was returned. \nFull response: {post_response.text}'
+        assert post_response.status_code == 200, 'Expected status code was 200, but {post_response.status_code} was ' \
+                                                 'returned. \nFull response: {post_response.text}'
```
Comment (Rebits) [inline tests/integration/test_api/test_config/test_use_only_authd/test_use_only_authd.py:171]:
```suggestion
        assert post_response.status_code == 200, f'Expected status code was 200, but {post_response.status_code} was ' \
                                                 f'returned. \nFull response: {post_response.text}'
```

**Review by Rebits — CHANGES_REQUESTED** [review]
GJ but some changes are required.

Due to it is necessary to document all tests, as they have been modified, it would be desirable to update the docstrings to follow repository standards.

File: tests/integration/test_api/test_config/test_bruteforce_blocking_system/test_bruteforce_blocking_system.py (hunk 1 above)
Comment (Rebits) [inline tests/integration/test_api/test_config/test_bruteforce_blocking_system/test_bruteforce_blocking_system.py:53]: Maybe is better to use a less generic exception.
```suggestion
    with pytest.raises(RuntimeError):
```

**Review by Rebits — CHANGES_REQUESTED** [review]

**Review by Rebits — APPROVED** [review]
LGTM

**Pull request merged**